chasm/lib/activity/statemachine.go

```diff
@@ -33,28 +40,120 @@ var TransitionScheduled = chasm.NewTransition(
 			return err
 		}
 
+		if attempt.Count != 0 {
```

Comment:
If you don't expect this to happen, don't check for it. This can only be due to a bug in this library.